html/cross-origin-embedder-policy/reporting-to-endpoint.https.html

```diff
@@ -77,100 +77,82 @@
   assert_unreached(`A report whose blockedURL is ${blockedUrl} and url is ${contextUrl} is not found.`);
 }
 
-async_test(async (t) => {
-  try {
-    const iframe = document.createElement('iframe');
-    t.add_cleanup(() => iframe.remove());
-
-    iframe.src = `resources/reporting-empty-frame.html`
-    document.body.appendChild(iframe);
-    await new Promise(resolve => {
-      iframe.addEventListener('load', resolve, {once: true});
-    });
-
-    const url = `${REMOTE_ORIGIN}/common/text-plain.txt?${token()}`;
-    const init = { mode: 'no-cors', cache: 'no-store' };
-    // The response comes from cross-origin, and doesn't have a CORP
-    // header, so it is blocked.
-    iframe.contentWindow.fetch(url, init).catch(() => {});
-
-    // Wait 3 seconds for reports to settle.
-    await wait(3000);
-
-    checkCorpReportExistence(reports, url, iframe.src, 'enforce');
-    checkCorpReportExistence(
+promise_test(async t => {
```

Comment:
https://wpt.fyi/results/html/cross-origin-embedder-policy/reporting-to-endpoint.https.html?label=pr_head&max-count=1&pr=25043 shows this test taking ~10s now, but it has `<meta name="timeout" content="long">` so that's OK...

The "Wait 3 seconds for reports to settle" are what made me go look. If there is a way to run faster in the passing case, that would be great.

Reply:
maybe we can do something similar to what was done for cross-origin-opener-policy: pass as soon as there's a matching report, fail if no report available after 3s:  https://github.com/web-platform-tests/wpt/blob/f7ce8205f9a99ce7242904fbcd7e051eec4a0c79/html/cross-origin-opener-policy/reporting/resources/dispatcher.js#L33

Reply:
This is a bit of a complex refactor, so I've put it together in https://github.com/web-platform-tests/wpt/pull/25502 to discuss there.

Reply:
Thanks @stephenmcgruer!